tlspuffin/src/tls/rustls/msgs/handshake.rs

```diff
@@ -40,7 +40,25 @@ macro_rules! declare_u16_vec (
     #[derive(Debug, Clone)]
     pub struct $name(pub Vec<$itemtype>);
 
-    atom_extract_knowledge!(TLSProtocolTypes, $name);
+    impl Extractable<TLSProtocolTypes> for $name {
```

Comment:
Is this related to the PR?

Reply:
Yes, in order to query the NamedGroup in the unit test, I added the abilty to extract NamedGroup from the client hello.